### PR TITLE
Fix #276

### LINKED ISSUE #276: IndentLinesDisable not work in defx
https://github.com/Shougo/defx.nvim/issues/154
Setting the g:indentLine_fileTypeExclude like following or IndentLinesDisable are not work in defx. The conceallevel and concealcursor  are not reseted properly. 
```
let g:indentLine_fileTypeExclude = ['defx']
```

### PR DESCRIPTION
**Pull request by richin13**
See https://github.com/Yggdroot/indentLine/issues/276#issuecomment-541486799 for rationale

### CODE REVIEW

**Comment by itsjef** [comment]
@richin13 This fixed the issue, may I know why was it closed?

**Comment by richin13** [comment]
The logic currently being used is a bit complex & hard to follow. The change I did introduced many new bugs :sweat_smile:
